File: SWEA/D2/16504. Gravity/Gravity.py
T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    N = int(input())
    arr = list(map(int, input().split()))
# N개의 0을 가지고 있는 리스트 result 생성
    result = [0]*N
# 반복하면서 자기자신보다 작은 값들일 때 갯수를 셈
    for i in range(N):
        cnt = 0
        for j in range(i +1, N):
            if arr[i] > arr[j]:
               cnt += 1
        result[i] = cnt
    answer = 0
# 최대값 찾기
    for num in range(len(result)):
        if result[num] > answer:
            answer = result[num]
    print(f'#{test_case} {answer}')

File: SWEA/D2/16504. Gravity/Gravity.java
import java.util.Scanner;
import java.io.FileInputStream;
class Solution
{
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();
		for(int test_case = 1; test_case <= T; test_case++)
		{
			int N = sc.nextInt();
			int[] arr = new int[N];
			for (int i=0; i<N; i++) {
				arr[i] = sc.nextInt();
			}
			int[] result = new int[N];
			for (int i=0;i<N;i++) {
				int cnt = 0;
				for (int j=i+1;j<N;j++) {
					if (arr[i] > arr[j]) {
						cnt ++;
					}
				}
				result[i] = cnt;
			}
			int answer = 0;
			for (int num =0; num<result.length;num++) {
				if(result[num] > answer) {
					answer = result[num];
				}
			}
			System.out.println("#"+ test_case + " " + answer);
		}
	}
}